include/quda.h

```diff
@@ -129,6 +130,7 @@ extern "C" {
     double mq3;
 
     double mu;    /**< Twisted mass parameter */
+    double tm_rho; /**< Hasenbusch mass shift applied like twisted mass to diagonal (but not inverse) */
```

Comment:
I absolutely wanted to follow this path in the first place but realised that unfortunately, `clover_rho` and `tm_rho` are used differently. `rho` is added to the clover term as a real number on the diagonal in the clover accessor (see `clover_field_order.h`). `tm_rho`, instead, is added in operators and comes the imaginary unit, a gamma_5 and a tau_3. Hence, instead of locating it in the clover accessor, the only place where it comes into play is in the call of `ApplyTwistedClover` in `DiracTwistedClover::DslashXpay` by shifting the twisted quark mass.

In this sense, the duplication isn't really between `clover_rho` and `tm_rho` but between `DiracTwistedClover` and `DiracCloverHasenbuschTwist`, the latter being basically Twisted Clover but not quite, if I understood correctly.

Reply:
The complication then arises in differentiating whether the proposed `hasenbusch_rho` should be added to the diagonal in the clover accessor or whether it should take the place of `tm_rho` for `DiracTwistedClover` or `mu` for `DiracCloverHasenbuschTwist`.

IIRC it was nontrivial to prevent the `add_rho=true` instance of the clover accessors being instantiated (and used in the operators) when `clover_rho` is set non-zero. I don't remember the logic chain that I followed to conclude this and I might well have the wrong impression.

Reply:
Hmm. Having `clover_rho != 0` -> `cloverPrecise->setRho(inv_param->clover_rho);` (`interface_quda`). This in turn leads to `rho(clover.Rho())` in the accessor constructor but, if it is instantiated with `add_rho==false`, this shouldn't have any effect.

One issue that we certainly have from the tmLQCD side is that we use `tm_rho` also for Hasenbusch preconditioning regular untwisted clover. In that case, we almost certainly would run into problems (but I should be able to work around this by using `TwistedClover` with zero mu for this case on the tmLQCD side).

I'm also not a fan of the lack of unification of these three very similar things... It should be noted that at least in principle, `clover_rho` and `tm_rho` (or, equivalently, `mu` of `DiractCloverHasenbuschTwist`) are different and one might argue that at least two of the three should be allowed to coexist.

Reply:
> Add to quda_fortran.F90

9d6b5c58b57e4eedbd2cd32fb80ff9d1516573a4